### ISSUE TITLE
Default provider architecture is not used for 'unknown' box architecture

### ISSUE THREAD

**Issue by alexgit2k** [hashicorp/vagrant#13312]
### Description

One of the architecture can be selected as "Default provider architecture" which is then marked with a star on the frontend. If no architecture is supplied (like with `vagrant init`) it looks like vagrant wants to fetch the box for architecture `unknown` and is not using the architecture which has been set as default.

Alma Linux has no box for the `unknown` architecture, therefore `vagrant init` is not working:
- https://app.vagrantup.com/almalinux/boxes/9/versions/9.3.20231118

Rocky Linux has a box for the `unknown` architecture, therefore `vagrant init` is working:
- https://app.vagrantup.com/rockylinux/boxes/9/versions/3.0.0

### Expected behavior

Provisioned box.

### Actual behavior

```
The box you're attempting to add has no available version that
matches the constraints you requested. Please double-check your
settings. Also verify that if you specified version constraints,
that the provider you wish to use is available for these constraints.

Box: almalinux/9
Address: https://vagrantcloud.com/api/v2/vagrant/almalinux/9
Constraints: 9.3.20231118
Available versions: 9.0.20220513, 9.0.20220531, 9.0.20220715, 9.0.20220802, 9.0.20220823, 9.0.20220830, 9.0.20221003, 9.0.20221102, 9.1.20221117, 9.2.20230513, 9.3.20231118
```

#### Vagrant version

2.4.0

#### Host operating system

Windows 10

#### Guest operating system

Almalinux 9, Rocky Linux 9

##### Not working
1. vagrant init --box-version 9.3.20231118 --minimal almalinux/9
2. vagrant up

##### Working
1. vagrant init --box-version =3.0.0 --minimal rockylinux/9
2. vagrant up

##### Not working
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "almalinux/9"
  config.vm.box_version = "9.3.20231118"
end
```

##### Working
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "rockylinux/9"
  config.vm.box_version = "=3.0.0"
end
```

**Comment by chrisroberts** [comment]
Hi there. This has the host operating system listed as Linux and the guest operating system listed as Windows. However, the `vagrant init` is setting up a Linux guest. Is this issue being encountered on Windows host systems?

**Comment by alexgit2k** [comment]
@chrisroberts My mistake, corrected it now in the issue. Happens on a Windows-host with Linux-guests.

**Comment by chrisroberts** [comment]
Thanks for the clarification. Closing as a dup of #13274 which is fixed by #13278. [This comment](https://github.com/hashicorp/vagrant/issues/13274#issuecomment-1771705244) includes a workaround. The 2.4.1 release will include the fix which will be out at the end of the week.

Cheers!

**Comment by alexgit2k** [comment]
Thank you very much Chris! I can confirm that with the changes in #13278 it is now working:
- Boxes with only `amd64`-architecture: `vagrant init --box-version 9.3.20231118 --minimal almalinux/9`
- Also older boxes with only `unknown`-architecture: `vagrant init --box-version 2004.01 --minimal centos/7`
